java/opendcs/src/main/java/decodes/cwms/database/CwmsOracleProvider.java

```diff
@@ -0,0 +1,261 @@
+package decodes.cwms.database;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.Call;
+import org.jdbi.v3.postgres.PostgresPlugin;
+import org.opendcs.spi.database.MigrationProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import decodes.cwms.CwmsTimeSeriesDb;
+import decodes.dbimport.DbImport;
+import decodes.launcher.Profile;
+import decodes.tsdb.ImportComp;
+import decodes.tsdb.TimeSeriesDb;
+import decodes.util.DecodesSettings;
+import ilex.util.EnvExpander;
+
+/**
+ * CwmsOracleProvider provides support for handling installation and updates of the OpenDCS-CWMS-Oracle
+ * schema.
+ */
+public class CwmsOracleProvider implements MigrationProvider
+{
+    private static final Logger log = LoggerFactory.getLogger(CwmsOracleProvider.class);
+    public static final String NAME = "CWMS-Oracle";
+
+    private Map<String,String> placeholders = new HashMap<>();
+    private static final List<MigrationProvider.MigrationProperty> properties = new ArrayList<>();
+
+    static
+    {
+        properties.add(
+            new MigrationProperty(
+                "CWMS_SCHEMA", String.class,
+                "How many tables should be used to partition numeric timeseries data."));
+        properties.add(
+            new MigrationProperty(
+                "CCP_SCHEMA", String.class,
+                "How many tables should be used to balance text timeseries data."));
+        properties.add(
+            new MigrationProperty(
+                "DEFAULT_OFFICE_CODE", Integer.class,
+                ""));
+        properties.add(
+            new MigrationProperty(
+                "DEFAULT_OFFICE", String.class,
+                ""));
+        properties.add(
+            new MigrationProperty(
+                "TABLE_SPACE_SPEC", String.class,
+                ""));
+    }
+
+    @Override
+    public String getName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public Map<String, String> getPlaceholderValues()
+    {
+        return Collections.unmodifiableMap(placeholders);
+    }
+
+    @Override
+    public void setPlaceholderValue(String name, String value)
+    {
+        placeholders.put(name,value);
+    }
+
+    @Override
+    public void registerJdbiPlugins(Jdbi jdbi)
+    {
+        //jdbi.installPlugin(new OraclePlugin());
+    }
+
+    @Override
+    public List<MigrationProperty> getPlaceHolderDescriptions()
+    {
+        return Collections.unmodifiableList(properties);
+    }
+
+    @Override
+    public void createUser(Jdbi jdbi, String username, String password, List<String> roles)
+    {
+        /*
```

Comment:
If the goal is a  no-op (all code is commented out) consider logging a message to warn nothing will happen here.